ivy/functional/backends/paddle/experimental/layers.py

```diff
@@ -69,7 +69,39 @@ def max_pool3d(
     data_format: str = "NDHWC",
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
+    if isinstance(strides, int):
+        strides = (strides, strides, strides)
+    elif len(strides) == 1:
+        strides = (strides[0], strides[0], strides[0])
+    if isinstance(kernel, int):
+        kernel = (kernel, kernel, kernel)
+    elif len(kernel) == 1:
+        kernel = (kernel[0], kernel[0], kernel[0])
+    if data_format == "NDHWC":
+        x = paddle.transpose(x, perm=[0, 2, 3, 4, 1])
+    x_shape = list(x.shape[2:])
+    pad_d = _handle_padding(x_shape[0], strides[0], kernel[0], padding)
+    pad_h = _handle_padding(x_shape[1], strides[1], kernel[1], padding)
+    pad_w = _handle_padding(x_shape[2], strides[2], kernel[2], padding)
+    x = paddle.nn.functional.pad(
+        x,
+        [
+            pad_w // 2,
+            pad_w - pad_w // 2,
+            pad_h // 2,
+            pad_h - pad_h // 2,
+            pad_d // 2,
+            pad_d - pad_d // 2,
+        ],
+        value=float("-inf"),
+        data_format="NDHWC",
```

Comment:
why is the data format constant here? I know it's handled in L80:81 but I don't get the reason

Reply:
Because paddle's pad has defualt data_format=`NCHW`, which doesn't support 5 dims Tesnor. Also,In other backends `NDHWC` is defualt or may be they handle themsolves but they don't have data_format argument in their pad api.